### DISCUSSION TITLE
My HTML pages are 680K each. Am I doing something wrong?

### DISCUSSION THREAD

**Discussion by dogweather** [imfing/hextra#253] [Q&A] [answered]
It looks like it's from repeated tailwind css classes. I've never used tailwind before. Is there some production post-process that's meant to DRY these up?

Here's an example page: https://forkful.ai/en/go/calculating-a-date-in-the-future-or-past/

I'll be reducing the page's size when I finish the sidebar design. But this is crazy.

**Comment by imfing** [comment]
> [forkful.ai/en/go/calculating-a-date-in-the-future-or-past](https://forkful.ai/en/go/calculating-a-date-in-the-future-or-past/)

I didn't see the page is taking 680K space. Can you provide more details?

**Reply by dogweather** [reply]
Sure:

```bash
curl https://forkful.ai/en/go/calculating-a-date-in-the-future-or-past/ > page.html
ls -lh page.html
```
Output:
```bash
-rw-r--r--   665K Jan 14 14:18 page.html
```

That's with the standard sidebar functionality. I'm hacking it to only display the relevant links and that reduces the page weight to 75K. Much better, but still 10x more than 7.5K or so which it'd be with semantically based CSS classes. Or, classes that roll up the common utilities.

**Reply by imfing** [reply, marked as answer]
Gotcha. 

Apparently the sidebar items classes are taking too much space. We could replace those repetitive ones with some common class name. I'll create an issue and address this.

**Reply by dogweather** [reply]
Hey, thanks!
